test/cppsuite/src/component/statistics_writer.cpp

```diff
@@ -26,18 +26,18 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include "perf_plotter.h"
+#include "statistics_writer.h"
 
 namespace test_harness {
 void
-perf_plotter::add_stat(const std::string &stat_string)
+statistics_writer::add_stat(const std::string &stat_string)
 {
     std::lock_guard<std::mutex> lg(_stat_mutex);
     _stats.push_back(stat_string);
 }
 
 void
-perf_plotter::output_perf_file(const std::string &test_name)
```

Comment:
Not a fan of this naming change as it implies the statistics_monitor and writer have something in common, whereas this class is really just a file output mechanism.

Reply:
The `statistics_writer` can collect statistics and output them to a file.
The `statistics_monitor` performs runtime as well as post-run checks on wt stats and fails if the limits are not respected.
They both deal with statistics, that's the only thing they have in common, hence the prefix.
If their respective header is better documented, could we stick with those names?

Reply:
Do we still fail on limits or do we just output those as evergreen plots? Can we call it json_plot_writer, that way it would be more specific? I know the original name `perf_plotter` was incorrect as it didn't plot charts but it certainly was more disconnected from the `runtime_monitor`. I feel like `statistics_monitor` could also be misleading as it may imply we're only checking WiredTIger statistics, how about `state_monitor` or something?

Reply:
Yes we can still perform checks on top of generating Evergreen plots. The checks are enabled through those [configuration items](https://github.com/wiredtiger/wiredtiger/blob/develop/dist/test_data.py#L99-L102).
I don't think we want the word `plot` in the name, we are just writing stuff in a file and `json` is too specific. Later on, we may need a different format, and writing into the `json` format becomes an option.
The `statistics_monitor` could effectively monitor states too and any other types of data, that's why we thought `statistics` would capture all of that.

Reply:
Ultimately we can combine all of this functionality into a single class that has the ability to both monitor a metric and write it to a file for plotting (or whatever) so I don't see it as a problem if they share the same prefix. While I don't think there's a danger of people assuming we mean WiredTiger statistics (statistics is an extremely common term), there's no reason we can't monitor other types of information (as Etienne pointed out above). How about `data_monitor` and `data_writer`? That covers basically everything, and works nicely if we wanted to define specialized child classes in the future.

Reply:
Yeah data_monitor and data_writer sound better to me. I'll be happy with that.

Reply:
I'm gonna change my mind here, I think we should just go with the statistics naming scheme. `data` could be confused with the actual data written by the test. Apologies for the confusion.

Reply:
I think `data` is better. Maybe you are overthinking it?